activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/process/ProcessRuntimeBPMNErrorReceivedIT.java

```diff
@@ -0,0 +1,231 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.spring.boot.process;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertNotNull;
+
+import org.activiti.api.process.model.ProcessInstance;
+import org.activiti.api.process.model.builders.ProcessPayloadBuilder;
+import org.activiti.api.process.model.events.BPMNErrorReceivedEvent;
+import org.activiti.api.process.runtime.ProcessRuntime;
+import org.activiti.api.runtime.shared.query.Page;
+import org.activiti.api.runtime.shared.query.Pageable;
+import org.activiti.spring.boot.process.listener.DummyBPMNErrorReceivedListener;
+import org.activiti.spring.boot.security.util.SecurityUtil;
+import org.activiti.spring.boot.test.util.ProcessCleanUpUtil;
+import org.assertj.core.groups.Tuple;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE)
+public class ProcessRuntimeBPMNErrorReceivedIT {
+
+    private static final String ERROR_BOUNDARY_EVENT_SUBPROCESS = "errorBoundaryEventSubProcess";
+    private static final String ERROR_START_EVENT_SUBPROCESS = "errorStartEventSubProcess";
+    private static final String ERROR_BOUNDARY_EVENT_CALLACTIVITY = "catchErrorOnCallActivity";  
+
+    @Autowired
+    private ProcessRuntime processRuntime;
+    
+    @Autowired
+    private SecurityUtil securityUtil;
+
+    @Autowired
+    private ProcessCleanUpUtil processCleanUpUtil;
+    
+    @Autowired
+    private DummyBPMNErrorReceivedListener listener;
+    
+    @Before
+    public void init() {
+        listener.clear();
+    }
+
+    @After
+    public void cleanUp(){
+        processCleanUpUtil.cleanUpWithAdmin();
+        listener.clear();
+    }
+
+    @Test
+    public void should_CatchSubProcessBoundaryErrorEvent_When_ErrorEndEvenThrown(){
+
+        //given
+        securityUtil.logInAs("user");
+
+        //when
+        ProcessInstance processInstance = processRuntime.start(
+                ProcessPayloadBuilder
+                        .start()
+                        .withProcessDefinitionKey(ERROR_BOUNDARY_EVENT_SUBPROCESS)
+                        .build());
+        
+        //then
+        assertNotNull(processInstance);
+        
+        //Error-handling should end the process
+        Page<ProcessInstance> processInstancePage = processRuntime.processInstances(Pageable.of(0,
```

Comment:
Here, and in the following tests, I think it would be interesting to have a user task following the error catch event, so you can test that the process is still running and it's pointing to the user task.

Reply:
I have modified test processes... Looks good.
But for on case I still have to figure out, what is going on:
![image](https://user-images.githubusercontent.com/21080352/64965883-11686f00-d89e-11e9-95ed-f967b5b3e2a0.png)

Reply:
The test is fixed now: all work like expected...